File: user-service/src/main/java/com/chat/kafka/producer/ElasticKafkaProducer.java
package com.chat.kafka.producer;

import com.chat.config.KafkaConfigData;
import com.chat.kafka.avro.model.UserAvroModel;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.kafka.core.reactive.ReactiveKafkaProducerTemplate;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Flux;

import java.util.Date;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class ElasticKafkaProducer {
    private final ReactiveKafkaProducerTemplate<String, UserAvroModel> userToElasticProducerTemplate;
    private final KafkaConfigData kafkaConfigData;

    public void run() {
        Flux.range(0, 1)
                .map(index -> UserAvroModel
                        .newBuilder()
                        .setId(UUID.randomUUID().toString())
                        .setName("random-name")
                        .setUsername("username-1")
                        .setCreatedDate(new Date().getTime())
                        .build()
                )
                .doOnNext(it -> System.out.println("Data: $it"))
                .flatMap(it -> this.userToElasticProducerTemplate.send(kafkaConfigData.getTopicName(), it.getId(), it))
                .doOnError(it -> System.out.printf("Error: %s%n", it.getMessage()))
                .doOnNext(it -> System.out.printf("Result:  %s%n", it.recordMetadata().toString()))
                .subscribe();
    }
}

File: kafka-to-elastic-service/src/main/kotlin/com/chat/producer/ElasticKafkaProducer.kt
package com.chat.producer

import com.chat.kafka.avro.model.UserAvroModel
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.boot.autoconfigure.kafka.KafkaProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.core.reactive.ReactiveKafkaProducerTemplate
import reactor.kafka.sender.SenderOptions
import java.util.*

//@Configuration
//class ElasticKafkaProducer {
//
//    @Bean("user-to-elastic-producer-option")
//    fun receiverOps(kafkaProperties: KafkaProperties): SenderOptions<String, UserAvroModel> {
//        return SenderOptions.create(kafkaProperties.buildProducerProperties())
//    }
//
//    @Bean("user-to-elastic-producer-template")
//    fun reactiveKafkaProducerTemplate(
//        @Qualifier("user-to-elastic-producer-option")
//        receiverOptions: SenderOptions<String, UserAvroModel>
//    ): ReactiveKafkaProducerTemplate<String, UserAvroModel> {
//        return ReactiveKafkaProducerTemplate(receiverOptions)
//    }
//}